pkg/pipelines/service.go

```diff
@@ -69,7 +69,12 @@ func serviceResources(m *config.Manifest, fs afero.Fs, p *AddServiceParameters)
 
 	cfg := m.GetPipelinesConfig()
 	if cfg != nil && p.WebhookSecret == "" && p.GitRepoURL != "" {
-		return nil, fmt.Errorf("The webhook secret is required")
+		gitSecret, err := util.GenerateSecureString(20)
+		if err != nil {
+			return nil, err
+		}
+		p.WebhookSecret = gitSecret
+
 	}
 
```

Comment:
Any reasons why `geenerate secret` is done here in the backend for `service add` command but `bootstrap` and `init` commands are in the cli?   In fact, this validation should probably have been done in the CLI's Validate().

Reply:
Yes William, the config retrieved form the manifest exists at the pkg/pipelines level.
The end to end testing is now complete.
Kindly review and approve changes.